File: 151051/Makovetskiy/task330+340+350/dc-rest/src/main/java/com/rmakovetskij/dc_rest/model/dto/requests/EditorRequestTo.java
package com.rmakovetskij.dc_rest.model.dto.requests;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class EditorRequestTo {

    private Long id;

    @NotBlank
    @Size(min = 2, max = 64)
    private String login;

    @NotBlank
    @Size(min = 8, max = 128)
    private String password;

    @NotBlank
    @Size(min = 2, max = 64)
    private String firstname;

    @NotBlank
    @Size(min = 2, max = 64)
    private String lastname;
}

File: 251002/Korti/publisher/src/providers/Editor/Dto/EditorRequestTo.ts
import { IsInt, IsString, Length } from 'class-validator';

export class EditorRequestTo {
  @IsString()
  @Length(2, 64)
  login: string;

  @IsString()
  @Length(8, 128)
  password: string;

  @IsString()
  @Length(2, 64)
  firstname: string;

  @IsString()
  @Length(2, 64)
  lastname: string;
}

export class UpdateEditorDto {
  @IsInt()
  id: number;

  @IsString()
  @Length(2, 64)
  login: string;

  @IsString()
  @Length(8, 128)
  password: string;

  @IsString()
  @Length(2, 64)
  firstname: string;

  @IsString()
  @Length(2, 64)
  lastname: string;
}
